pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java

```diff
@@ -105,6 +108,38 @@ public ResultSetGroup execute(Request request)
     return execute(null, request);
   }
 
+  /**
+   * Extract tableName from the given query
+   * @param query
+   * @return tableName
+   * @throws PinotClientException 
+   */
+  public String getTableNameFromQuery(String query) {
+    PinotQuery pinotQuery;
+
+    try {
+      pinotQuery = CalciteSqlParser.compileToPinotQuery(query);
+    } catch (Exception e) {
+      LOGGER.info("Caught exception while compiling SQL request: {}, {}", query, e.getMessage());
+      throw new PinotClientException("Caught exception while compiling SQL request");
+    }
+
+    DataSource dataSource;
+    
+    try {
+      dataSource = pinotQuery.getDataSource();
+    } catch (Exception e) {
+      throw new PinotClientException("Caught exception while extracting DataSource from PinotQuery. DataSource is null.");
+    }
+    
+    String tableName = dataSource.getTableName();
+    
+    if(tableName == null) {
```

Comment:
If tableName is null, then return null.

I'm thinking of the case like `select 1` which is a valid SQL and should be executed by any pinot broker.

Reply:
Thanks for the suggestion! I have fixed this and added the appropriate test for this scenario.